### PR TITLE
feat(dws): add datasource to query snapshot statistics of the cluster

### PR DESCRIPTION
**Pull request by wuzhuanhong**
**What this PR does / why we need it**:

Add new data source to query snapshot statistics under the specified DWS cluster.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. add new data source.
2. add realted to document and acceptance test.
```

## PR Checklist

* [x] Tests added/passed.

```
make testacc TEST='./huaweicloud/services/acceptance/dws' TESTARGS='-run TestAccDataSourceDwsClusterSnapshotStatistics_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dws -v -run TestAccDataSourceDwsClusterSnapshotStatistics_basic -timeout 360m -parallel 4
=== RUN   TestAccDataSourceDwsClusterSnapshotStatistics_basic
=== PAUSE TestAccDataSourceDwsClusterSnapshotStatistics_basic
=== CONT  TestAccDataSourceDwsClusterSnapshotStatistics_basic
--- PASS: TestAccDataSourceDwsClusterSnapshotStatistics_basic (337.86s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dws       338.007s
```

* [x] Documentation updated.
* [x] Schema updated.
* [ ] CheckDeleted.

  - **a. During query operation (Read Context)**
    aa. Resource not found
    \>>>>>> Paste the screenshot here <<<<<<

    
  - **b. During delete/disassociate/unbind operation (Delete Context)**
    ba. Resource not found
    \>>>>>> Paste the screenshot here <<<<<<

### CODE REVIEW

File: huaweicloud/services/acceptance/dws/data_source_huaweicloud_dws_cluster_snapshot_statistics_test.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+}
+	
+data "huaweicloud_dws_cluster_snapshot_statistics" "test" {
+  depends_on = [huaweicloud_dws_snapshot.test]
+  cluster_id = "%[1]s"
+}
+
+locals {
+  statistics   = data.huaweicloud_dws_cluster_snapshot_statistics.test.statistics
+  free_capaticy = [for v in local.statistics : v if v.name == "storage.free"]
```
Comment (profoundwu) [inline huaweicloud/services/acceptance/dws/data_source_huaweicloud_dws_cluster_snapshot_statistics_test.go]:
```suggestion
  statistics    = data.huaweicloud_dws_cluster_snapshot_statistics.test.statistics
  free_capaticy = [for v in local.statistics : v if v.name == "storage.free"]
```

**Comment by wuzhuanhong** [comment]
make testacc TEST='./huaweicloud/services/acceptance/dws' TESTARGS='-run TestAccDataSourceDwsClusterSnapshotStatistics_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dws -v -run TestAccDataSourceDwsClusterSnapshotStatistics_basic -timeout 360m -parallel 4
=== RUN   TestAccDataSourceDwsClusterSnapshotStatistics_basic
=== PAUSE TestAccDataSourceDwsClusterSnapshotStatistics_basic
=== CONT  TestAccDataSourceDwsClusterSnapshotStatistics_basic
--- PASS: TestAccDataSourceDwsClusterSnapshotStatistics_basic (345.72s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dws       345.784s

File: huaweicloud/services/acceptance/dws/data_source_huaweicloud_dws_cluster_snapshot_statistics_test.go [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  cluster_id = "%[1]s"
+}
+
+locals {
+  statistics    = data.huaweicloud_dws_cluster_snapshot_statistics.test.statistics
+  free_capaticy = [for v in local.statistics : v if v.name == "storage.free"]
+  used_capaticy = [for v in local.statistics : v if v.name == "storage.used"]
+}
+
+output "is_total_capacity" {
```
Comment (Lance52259) [inline huaweicloud/services/acceptance/dws/data_source_huaweicloud_dws_cluster_snapshot_statistics_test.go]:
```suggestion
output "is_total_capacity_set" {
```

**Comment by wuzhuanhong** [comment]
make testacc TEST=./huaweicloud/services/acceptance/dws TESTARGS='-run TestAccDataSourceDwsClusterSnapshotStatistics_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dws -v -run TestAccDataSourceDwsClusterSnapshotStatistics_basic -timeout 360m -parallel 4
=== RUN   TestAccDataSourceDwsClusterSnapshotStatistics_basic
=== PAUSE TestAccDataSourceDwsClusterSnapshotStatistics_basic
=== CONT  TestAccDataSourceDwsClusterSnapshotStatistics_basic
--- PASS: TestAccDataSourceDwsClusterSnapshotStatistics_basic (372.80s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dws       372.864s

File: huaweicloud/services/acceptance/dws/data_source_huaweicloud_dws_cluster_snapshot_statistics_test.go (hunk 2 above)
Comment (wuzhuanhong) [reply huaweicloud/services/acceptance/dws/data_source_huaweicloud_dws_cluster_snapshot_statistics_test.go]: fixed

**Comment by wuzhuanhong** [comment]
make testacc TEST=./huaweicloud/services/acceptance/dws TESTARGS='-run TestAccDataSourceDwsClusterSnapshotStatistics_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dws -v -run TestAccDataSourceDwsClusterSnapshotStatistics_basic -timeout 360m -parallel 4
=== RUN TestAccDataSourceDwsClusterSnapshotStatistics_basic
=== PAUSE TestAccDataSourceDwsClusterSnapshotStatistics_basic
=== CONT TestAccDataSourceDwsClusterSnapshotStatistics_basic
--- PASS: TestAccDataSourceDwsClusterSnapshotStatistics_basic (372.80s)
PASS
ok github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dws 372.864s

**Pull request merged**